screeps-game-api/src/constants.rs

```diff
@@ -233,17 +217,7 @@ pub enum Direction {
     TopLeft = 8,
 }
 
-impl TryFrom<Value> for Direction {
-    type Error = ConversionError;
-
-    fn try_from(v: Value) -> Result<Self, Self::Error> {
-        let as_num = u32::try_from(v)?;
-
-        Ok(Self::from_u32(as_num).unwrap_or_else(|| {
-            panic!("encountered a direction code we don't know: {}", as_num);
-        }))
-    }
-}
+js_deserializable!(Direction);
```

Comment:
I thought we had agreed not to use `js_deserializable!` since it was too costly?

Reply:
I think it should be better with `stdweb` 0.4.9+! Deserializing now no longer uses json parsing, and instead implements `TryFrom<Value>` by traversing the value with serde, calling into javascript to get any references.

I think it should be fairly equivalent to a manual implementation of `TryFrom` now, especially with these integer constants which should just be represented by `Value::Number`.

Reply:
Oh, that's great news, I didn't know!

Reply:
it is! I should have said something though rather than just barging ahead.